clusterloader2/pkg/measurement/common/network/network_performance_measurement.go

```diff
@@ -0,0 +1,474 @@
+			return true, nil
+		}
+	}
+	return false, err
+}
+
+// getWeightedTimerValuesForPoll returns poll timeout value.
+// Pod numbers are taken into consideration to decide the timer values.
+func (npm *networkPerformanceMeasurement) getWeightedPodReadyTimeout() time.Duration {
+	weightedPodReadyTimeout := float32(npm.podReplicas) * podReadyTimeOutFactor
+	podReadyTimeout := time.Duration(weightedPodReadyTimeout) * time.Second
+
+	klog.Info("podReadyTimeout: ", podReadyTimeout)
+	return podReadyTimeout
+
+}
+
+func (*networkPerformanceMeasurement) String() string {
+	return networkPerformanceMetricsName
+}
+
+func (npm *networkPerformanceMeasurement) storeWorkerPods() {
+	var podsWithIPs int
+	options := metav1.ListOptions{}
+	pods, _ := npm.k8sClient.CoreV1().Pods(netperfNamespace).List(context.TODO(), options)
+	for _, pod := range pods.Items {
+		if pod.Status.PodIP != "" {
+			podData := &workerPodData{podName: pod.Name, podIP: pod.Status.PodIP, workerNode: pod.Spec.NodeName}
+			npm.populateWorkerPodList(podData)
+			podsWithIPs++
+		}
+	}
+	klog.Infof("Pods configured: %v, Pods with IPs: %v", npm.podReplicas, podsWithIPs)
+}
+
+func (npm *networkPerformanceMeasurement) gather() (measurement.Summary, error) {
+	npm.createResultSummary()
+	content, err := util.PrettyPrintJSON(&measurementutil.PerfData{
+		Version:   "v1",
+		DataItems: npm.resultSummary.DataItems,
+	})
+	if err != nil {
+		klog.Info("Pretty Print to Json Err: ", err)
+	}
+	summaryName := fmt.Sprintf(npm.String() + npm.resultSummary.podRatio + npm.resultSummary.protocol + npm.resultSummary.service)
+	return measurement.CreateSummary(summaryName, "json", content), nil
+}
+
+// Dispose disposes resources,objects after the measurement.
+func (npm *networkPerformanceMeasurement) Dispose() {
+	if npm.framework == nil {
+		klog.V(1).Infof("Network measurement %s wasn't started, skipping the Dispose() step", npm)
+		return
+	}
+	klog.Infof("Stopping network performance measurement...")
+	if err := npm.deleteCRD(); err != nil {
+		klog.Errorf("error while deleting CRD: %v", err)
+	}
+
+	if err := npm.deleteClusterRoleBinding(); err != nil {
+		klog.Errorf("error while deleting rbac: %v", err)
+	}
+	if err := client.DeleteNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		klog.Errorf("error while deleting %s namespace: %v", netperfNamespace, err)
+	}
+	if err := client.WaitForDeleteNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		klog.Errorf("error while waiting for %s namespace to be deleted: %v", netperfNamespace, err)
+	}
+
+	npm.closeCh()
+}
+
+func (npm *networkPerformanceMeasurement) watchNetworkTestEvents() {
+	dynamicClient := npm.framework.GetDynamicClients().GetClient()
+	gvk := schema.GroupVersionKind{Group: customResourceGroup, Kind: customResourceKind, Version: customResourceVersion}
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	watchResource, err := dynamicClient.Resource(gvr).Watch(context.TODO(), metav1.ListOptions{})
```

Comment:
As I already mentioned - don't use watch yourself. You don't handle watch restarts, timeout, and so on.

Please use informer.

Reply:
Modified as per suggestion.